src/Build/BuildCheck/Infrastructure/BuildCheckBuildEventHandler.cs

```diff
@@ -20,15 +21,19 @@ internal class BuildCheckBuildEventHandler
 
     private readonly Dictionary<Type, Action<BuildEventArgs>> _eventHandlers;
 
+    private IDictionary<int, bool> _projectRestore;
+
     internal BuildCheckBuildEventHandler(
         IAnalysisContextFactory analyzerContextFactory,
         IBuildCheckManager buildCheckManager)
     {
         _buildCheckManager = buildCheckManager;
         _analyzerContextFactory = analyzerContextFactory;
 
+        _projectRestore = new Dictionary<int, bool>();
```

Comment:
This feels overcomplicated - we should be able to skip everything if BuildSubmissionStarted indicates restore and resume if next BuildSubmissionStarted indicates otherwise